File: app/lib/features/medical/domain/medical_phrase.dart
/// A medical phrase with Japanese text, reading, and translation.
class MedicalPhrase {
  const MedicalPhrase({
    required this.id,
    required this.category,
    required this.phraseJa,
    this.phraseReading,
    required this.translation,
    this.context,
  });

  final String id;
  final String category;
  final String phraseJa;
  final String? phraseReading;
  final String translation;
  final String? context;

  factory MedicalPhrase.fromJson(Map<String, dynamic> json) {
    return MedicalPhrase(
      id: json['id'] as String,
      category: json['category'] as String? ?? '',
      phraseJa: json['phrase_ja'] as String? ?? '',
      phraseReading: json['phrase_reading'] as String?,
      translation: json['translation'] as String? ?? '',
      context: json['context'] as String?,
    );
  }
}

File: backend/app_service/models/medical_phrase.py
"""MedicalPhrase model — maps to the medical_phrases table (DATA_MODEL.md §15)."""

import json
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, SmallInteger, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from database import Base


class MedicalPhrase(Base):
    """Medical phrase for symptom translation — master data."""

    __tablename__ = "medical_phrases"

    id: Mapped[str] = mapped_column(
        String(36), primary_key=True, default=lambda: str(uuid.uuid4())
    )
    category: Mapped[str] = mapped_column(String(30), nullable=False)
    phrase_ja: Mapped[str] = mapped_column(Text, nullable=False)
    phrase_reading: Mapped[str | None] = mapped_column(
        Text, nullable=True, default=None
    )
    translations: Mapped[str] = mapped_column(Text, nullable=False)  # JSONB
    context: Mapped[str | None] = mapped_column(
        Text, nullable=True, default=None
    )  # JSONB
    sort_order: Mapped[int] = mapped_column(SmallInteger, nullable=False, default=0)
    is_active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, server_default=func.now(),
        onupdate=func.now(),
    )

    # ── JSONB helper methods ───────────────────────────────────────────

    def get_translation(self, lang: str = "en") -> str:
        data = json.loads(self.translations) if isinstance(self.translations, str) else self.translations
        if isinstance(data, dict):
            return data.get(lang, data.get("en", ""))
        return ""

    def get_context(self, lang: str = "en") -> str | None:
        if not self.context:
            return None
        data = json.loads(self.context) if isinstance(self.context, str) else self.context
        if isinstance(data, dict):
            return data.get(lang, data.get("en"))
        return None
